lib/worker/monitor_ets.ex

```diff
@@ -0,0 +1,90 @@
+defmodule ExRabbitPool.Worker.MonitorEts do
+  use GenServer
+
+  require Logger
+
+  @name __MODULE__
+  @tab :monitors_tab
+
+  #########
+  ## API
+  #########
+
+  def start_link(opts \\ []) do
+    GenServer.start_link(@name, :ok, opts ++ [name: @name])
+  end
+
+  @doc false
+  def get_monitors do
+    monitors = monitors()
+  end
+
+  @doc false
+  def add(monitor) do
+    monitors_ets = monitors()
+    true = :ets.insert(@tab, {:monitors, [monitor|monitors_ets]})
+  end
+
+  @doc false
+  def remove_monitor(pid) do
+    monitors_ets = monitors()
+    remove_monitor(monitors_ets, pid)
+  end
+
+  ######################
+  ## Server Callbacks
+  ######################
+
+  @impl true
+  def init(:ok) do
+    :ets.new(@tab, [:set, :named_table, :public, read_concurrency: true,
+                                                 write_concurrency: true])
+    true = :ets.insert(@tab, {:monitors, []})
+    {:ok, %{}}
+  end
+
+  @impl true
+  def handle_info(msg, state) do
+    Logger.error("Unexpected message at MonitorEts: #{msg}")
+    {:noreply, state}
+  end
+
+  ##############
+  ## Internal
+  ##############
+
+  defp monitors do
+    [monitors: monitors_ets] = :ets.lookup(@tab, :monitors)
+    monitors_ets
+  end
+
+  defp remove_monitor(monitors, client_ref) when is_reference(client_ref) do
+    monitors
```

Comment:
you should be using `:ets.select_delete/2` http://erlang.org/doc/man/ets.html#select_delete-2 or something similar for this, instead of deleting records manually from the ets table

Reply:
ok, I'll change this.